feat: Harmonize app dialogs

g123k opened the pull request:
On Android & iOS, we should have Dialogs with buttons in the following order:
Negative action - Neutral action - Positive action

This PR ensure will now follow this rule.
Also, if the device uses an RTL language, actions will be reversed.

Will fix #1955

g123k committed:
Harmonize app dialogs

monsieurtanuki commented on packages/smooth_app/lib/generic_lib/dialogs/smooth_alert_dialog.dart:
```diff
@@ -2,23 +2,25 @@ import 'package:flutter/material.dart';
-///	Open by calling
-///
-///showDialog<void>(
+/// showDialog<void>(
```
```suggestion
/// ```dart
/// showDialog<void>(
```

monsieurtanuki commented on packages/smooth_app/lib/generic_lib/dialogs/smooth_alert_dialog.dart:
```diff
@@ -2,23 +2,25 @@ import 'package:flutter/material.dart';
 ///        builder: (BuildContext context) {
 ///          return SmoothAlertDialog(...)
 ///	}
 ///)
```
```suggestion
///)
/// ```
```

monsieurtanuki commented on packages/smooth_app/lib/pages/preferences/user_preferences_faq.dart:
```diff
@@ -117,28 +117,26 @@ class UserPreferencesFaq extends AbstractUserPreferences {
+            );
+          },
+          text: appLocalizations.licenses,
+          minWidth: 100,
```
I guess we could get rid of all those `minWidth: 100`, by handling it in `SmoothAlertDialog`.
It's not like "for this specific dialog, we have this specific need", it's just that in all cases we want a decent button size.

monsieurtanuki reviewed (approved):
Thank you @g123k, that's great!
Please have a look at my comments.

A suggestion: when there are two buttons like Cancel and OK, display Cancel as a mere `TextButton` and OK as a more triumphant `ElevatedButton`.

g123k commented:
That would indeed be a nice idea, I will make the changes for that.
In the case, where we have three buttons, which Widget should I use for the neutral/middle one?

monsieurtanuki commented:
I can't remember having seen an example of 3 buttons in your code.
Anyway, let's say that the TextButton would only be for the non-positive button when in combination with a positive button.

g123k committed:
Flat/text button for negative & neutral states
+ fix a missing image

g123k commented:
I changed a bit the implementation to only provide a `SmoothActionButton` and the `Dialog` will "decide" which design to use.
I have also fixed an issue with the FAQ dialog, where the icon was linked to a missing resource.

Note: your suggestions are directly applied in my last commit

monsieurtanuki commented on packages/smooth_app/lib/pages/preferences/user_preferences_faq.dart:
```diff
@@ -75,7 +75,9 @@ class UserPreferencesFaq extends AbstractUserPreferences {
             ListTile(
-              leading: Image.asset('assets/app/smoothie-icon.1200x1200.png'),
+              leading: SvgPicture.asset(
+                'assets/app/release_icon.svg',
```
I would use a local `const String` as you're using it twice.

g123k replied:
Applied in my last commit :)

monsieurtanuki reviewed (changes requested):
Hi @g123k!
I think having two implementations of `SmoothActionButton` in distinct files is not brillant.
What about `SmoothAction` for your new field-container class?

g123k committed:
Remove unused SmoothActionButton implementation

g123k commented:
Sorry, It was a mistake, the old implementation is not used anymore

g123k committed:
Fix build issues

teolemon commented:
I'm merging, thanks @g123k

g123k commented:
No no just a second, there is an issue.
I'll fix it ASAP
